pyccl/halos/pk_hm.py

```diff
@@ -0,0 +1,569 @@
+from .. import ccllib as lib
+from .hmfunc import MassFunc
+from .hbias import HaloBias
+from .profiles import HaloProfile
+from ..core import check
+from ..pk2d import Pk2D
+from ..power import linear_matter_power, nonlin_matter_power
+from ..background import rho_x
+from ..pyutils import _spline_integrate
+import numpy as np
+
+
+class ProfileCovar(object):
+    """ This class implements the 1-halo covariance between two
+    halo profiles. In the simplest case, this covariance is just
+    the product of both profiles in Fourier space.
+    More complicated cases should be implemented by subclassing
+    this class and overloading the :meth:`~ProfileCovar.fourier_covar`
+    method.
+    """
+    def __init__(self):
+        pass
+
+    def fourier_covar(self, prof, cosmo, k, M, a,
+                      prof_2=None, mass_def=None):
+        """ Returns the Fourier-space two-point moment between
+        two profiles:
+
+        .. math::
+           \\langle\\rho_1(k)\\rho_2(k)\\rangle.
+
+        Args:
+            prof (:class:`~pyccl.halos.profiles.HaloProfile`):
+                halo profile for which the second-order moment
+                is desired.
+            cosmo (:class:`~pyccl.core.Cosmology`): a Cosmology object.
+            k (float or array_like): comoving wavenumber in Mpc^-1.
+            M (float or array_like): halo mass in units of M_sun.
+            a (float): scale factor.
+            prof_2 (:class:`~pyccl.halos.profiles.HaloProfile`):
```

Comment:
This underscore is funny.

Reply:
the one in the name.

Reply:
Do you mean you'd prefer `prof2` to `prof_2`? All done in that case.